staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,458 @@
+		if flag.Shorthand != "" {
+			allShorthands[flag.Shorthand] = struct{}{}
+		}
+	})
+
+	for _, fl := range aliasArgs.flags {
+		existingFlag := foundAliasCmd.Flag(fl.Name)
+		if existingFlag == nil {
+			return args, fmt.Errorf("invalid alias flag %s in alias %s", fl.Name, args[0])
+		}
+		if searchInArgs(existingFlag.Name, existingFlag.Shorthand, allShorthands, args) {
+			// Don't modify the value implicitly, if it is passed in args explicitly
+			continue
+		}
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			return args, fmt.Errorf("could not apply value %s to flag %s in alias %s err: %w", fl.Default, fl.Name, args[0], err)
+		}
+	}
+
+	if len(aliasArgs.prependArgs) > 0 {
+		// prependArgs defined in kuberc should be inserted after the alias name.
+		if commandIndex+1 >= len(args) {
+			// command is the last item, we simply append just like appendArgs
+			args = append(args, aliasArgs.prependArgs...)
+		} else {
+			args = append(args[:commandIndex+1], append(aliasArgs.prependArgs, args[commandIndex+1:]...)...)
+		}
+	}
+	if len(aliasArgs.appendArgs) > 0 {
+		// appendArgs defined in kuberc should be appended to actual args.
+		args = append(args, aliasArgs.appendArgs...)
+	}
+	// Cobra (command.go#L1078) appends only root command's args into the actual args and ignores the others.
+	// We are appending the additional args defined in kuberc in here and
+	// expect that it will be passed along to the actual command.
+	rootCmd.SetArgs(args[1:])
+	return args, nil
+}
+
+// DefaultGetPreferences returns KubeRCConfiguration.
+// If users sets kuberc file explicitly in --kuberc flag, it has the highest
+// priority. If not specified, it looks for in KUBERC environment variable.
+// If KUBERC is also not set, it falls back to default .kuberc file at the same location
+// where kubeconfig's defaults are residing in.
+// If KUBERC is set to "off", kuberc will be turned off and original behaviors in kubectl will be applied.
+func DefaultGetPreferences(kuberc string, errOut io.Writer) (*config.Preference, error) {
+	if val := os.Getenv("KUBERC"); val == "off" {
+		if kuberc != "" {
+			return nil, fmt.Errorf("disabling kuberc via KUBERC=off and passing kuberc flag are mutually exclusive")
+		}
+		return nil, nil
+	}
+	kubeRCFile := RecommendedKubeRCFile
+	explicitly := false
+	if kuberc != "" {
+		kubeRCFile = kuberc
+		explicitly = true
+	}
+
+	if kubeRCFile == "" && os.Getenv("KUBERC") != "" {
+		kubeRCFile = os.Getenv("KUBERC")
+		explicitly = true
+	}
+
+	preference, err := decodePreference(kubeRCFile)
+	switch {
+	case explicitly && preference != nil && runtime.IsStrictDecodingError(err):
+		// if explicitly requested, just warn about strict decoding errors if we got a usable Preference object back
+		fmt.Fprintf(errOut, "kuberc error: %v", err)
+		return preference, nil
+
+	case explicitly && err != nil:
+		// if explicitly requested, error on any error other than a StrictDecodingError
+		return nil, err
```

Comment:
wrap / prefix with `kuberc:`?

```suggestion
		return nil, fmt.Errorf("kuberc: %w", err)
```

Reply:
Updated.